nilearn/_utils/estimator_checks.py

```diff
@@ -3643,6 +3652,11 @@ def check_masker_generate_report(estimator):
     _check_html(report, is_fit=False)
     assert "Make sure to run `fit`" in str(report)
 
+    assert (
```

Comment:
Am pretty sure those asserts can be moved almost at the top of this function, as it must receive an initialized masker as input.